component/core/src/main/java/org/exoplatform/social/core/listeners/GroupSynchronizationSocialProfileListener.java

```
@@ -0,0 +1,164 @@
+                                                                          .map(ProfilePropertySetting::getPropertyName)
+                                                                          .toList();
+      Set<Map.Entry<String, Object>> properties = profile.getProperties().entrySet();
+      properties.stream().filter(property -> synchronizedProperties.contains(property.getKey())).forEach(property -> {
+        String propertyName = property.getKey();
+        String propertyValue = (String) property.getValue();
+        Group newPropertyNameGroup = createGroup(propertyName, profileGroup, organizationService);
+        if (newPropertyNameGroup != null) {
+          Group newPropertyValueGroup = createGroup(propertyValue, newPropertyNameGroup, organizationService);
+          addUserToGroup(newPropertyValueGroup, user, organizationService);
+        }
+      });
+    }
+  }
+
+  private Group getGroup(String groupId, OrganizationService organizationService) {
+    try {
+      GroupHandler groupHandler = organizationService.getGroupHandler();
+      return groupHandler.findGroupById(groupId);
+    } catch (Exception e) {
+      return null;
+    }
+  }
+
+  private Group createGroup(String groupName, Group parentGroup, OrganizationService organizationService) {
+    try {
+      Group group = getGroup(buildGroupId(parentGroup, groupName), organizationService);
+      if (group != null) {
+        return group;
+      }
+      GroupHandler groupHandler = organizationService.getGroupHandler();
+      Group newGroup = groupHandler.createGroupInstance();
+      newGroup.setGroupName(groupName.toLowerCase());
+      newGroup.setLabel(StringUtils.capitalize(groupName));
+      newGroup.setDescription(groupName + " group");
+      groupHandler.addChild(parentGroup, newGroup, true);
+      return getGroup(buildGroupId(parentGroup, groupName), organizationService);
+    } catch (Exception e) {
+      return null;
```

same remark as previous about error swallow + throwing for caller to be noticed + logging

---

same as https://github.com/Meeds-io/social/pull/2282#discussion_r1095431195